sdk/core/core/internal/az_clock_internal.h

```diff
@@ -13,7 +15,7 @@ AZ_INLINE uint64_t _az_clock_msec() {
   // Convert clock_t to milliseconds
   // Floating point arithmetic is used to cover CLOCKS_PER_SEC all values
   // including 12000, 2000, 1500, 300, 500, 100, 2
-  return (uint64_t)(clock() / (CLOCKS_PER_SEC / 1000.0));
+  return (uint64_t)(clock() / (CLOCKS_PER_SEC / (double)_az_TIME_MILLISECONDS_PER_SECOND));
```

Comment:
I've seen one libc for ARM where clock() wasn't properly connected to the sys-tick timer (always 0xFF..FF) - (Might have been libc from Arm GCC and FreeRTOS.)
Documenting the C99 requirement for time.h might be sufficient instead of extracting it to a PAL.

Reply:
On second thought - clock() may not have sufficient resolution for what you're using it.
Also, forcing uint64_t on an 8bit MCU is not a good idea.

Reply:
These sources are based on the old implementation, we have the PAL layer now, I am going to use that. The other question is that we may still redesign the PAL layer to use an equivalent of clock_t convertible to milliseconds, instead of pure milliseconds as we do now; but that would be a separate commit/PR.